clippy_lints/src/matches.rs

```diff
@@ -1237,6 +1237,22 @@ fn check_match_single_binding<'a>(cx: &LateContext<'a>, ex: &Expr<'a>, arms: &[A
     if in_macro(expr.span) || arms.len() != 1 || is_refutable(cx, arms[0].pat) {
         return;
     }
+
+    // Check that the match doesn't contain other branches that might be excluded by `#[cfg]` or other
+    // macros
```

Comment:
I think we should state more clearly in the comment that this is a `HACK:` and that we can't do better currently.